src/mlpack/methods/neat/neat_impl.hpp

```diff
@@ -0,0 +1,371 @@
+              newConnGeneList[j].setEnabled() = false;
+            else
+              newConnGeneList[j].setEnabled() = true;
+          }
+          // Weights will be assigned randomly in matching genes.
+          if (arma::randu<double>() < 0.5)
+            newConnGeneList[j].setWeight() = lessFitGenome.connectionGeneList[i]
+                .getWeight();
+          k = j;
+          break;
+        }
+      }
+    }
+
+    if (isAcyclic)
+    {
+      return Genome<ActivationFunction>(inputNodeCount, outputNodeCount,
+        newConnGeneList, nodeDepths, nextNodeID, bias, weightMutationProb,
+        weightMutationSize, biasMutationProb, biasMutationSize, 
+        nodeAdditionProb, connAdditionProb, isAcyclic);
+    }
+    else
+    {
+      return Genome<ActivationFunction>(inputNodeCount, outputNodeCount,
+        newConnGeneList, nextNodeID, bias, weightMutationProb,
+        weightMutationSize, biasMutationProb, biasMutationSize, 
+        nodeAdditionProb, connAdditionProb, isAcyclic);
+    }
+  }
+  else
+  {
+    size_t i = 0, j = 0;
+    size_t gen1size = gen1.connectionGeneList.size();
+    size_t gen2size = gen2.connectionGeneList.size();
+    size_t maxSize = gen1size > gen2size ? gen1size : gen2size;
+    size_t minSize = gen1size < gen2size ? gen1size : gen2size;
+    Genome<ActivationFunction>& maxGenome = gen1size > gen2size ? gen1 : gen2;
+    Genome<ActivationFunction>& minGenome = gen1size < gen2size ? gen1 : gen2;
+    while (j < minSize)
+    {
+      size_t innovID1 = maxGenome.connectionGeneList[i].getInnovationID();
+      size_t innovID2 = minGenome.connectionGeneList[j].getInnovationID();
+      if (innovID2 < innovID1)
+      {
+        if (arma::randu<double>() < 0.5)
+          newConnGeneList.push_back(minGenome.connectionGeneList[j++]);
+      }
+      else if (innovID2 == innovID1)
+      {
+        if (arma::randu<double>() < 0.5)
+          newConnGeneList.push_back(minGenome.connectionGeneList[j]);
+        else
+          newConnGeneList.push_back(maxGenome.connectionGeneList[i]);
+        i++;
+        j++;
+      }
+      else
+      {
+        if (arma::randu<double>() < 0.5)
+          newConnGeneList.push_back(maxGenome.connectionGeneList[i++]);
+      }
+    }
+    while (i < maxSize)
+    {
+      if (arma::randu<double>() < 0.5)
+        newConnGeneList.push_back(maxGenome.connectionGeneList[i++]);
+    }
+
+    size_t nextNodeID = gen1.nextNodeID > gen2.nextNodeID ? gen1.nextNodeID : gen2.nextNodeID;
+
+    return Genome<ActivationFunction>(inputNodeCount, outputNodeCount,
+          newConnGeneList, nextNodeID, bias, weightMutationProb,
+          weightMutationSize, biasMutationProb, biasMutationSize, 
+          nodeAdditionProb, connAdditionProb, isAcyclic);
+  }
+}
+
+template <class TaskType,
+          class ActivationFunction,
+          class SelectionPolicy>
+bool NEAT<TaskType, ActivationFunction, SelectionPolicy>::
+    compareGenome(Genome<ActivationFunction> gen1,
```

Comment:
As I mentioned earlier, always try using references, wherever applicable.

Reply:
Ah yes, i sometimes forget. I'll make the changes, thanks!